db/dbtypes/conversion.go

```diff
@@ -91,3 +91,20 @@ func ChartGroupingToInterval(grouping ChartGrouping) (float64, error) {
 func CalculateHashRate(difficulty, targetTimePerBlock float64) float64 {
 	return ((difficulty * math.Pow(2, 32)) / targetTimePerBlock) / 1000000
 }
+
+// CalculateWindowIndex calculates the window index from the quotient of a block
+// height and the chainParams.StakeDiffWindowSize. A window being a group of
+// blocks whose count is defined by chainParams.StakeDiffWindowSize, the first
+// window starts from block 1 to block 144 instead of block 0 to block 143.
+// To obtain the accurate window index value, we should add 1 to the quotient
+// obtained by dividing the block height with the chainParams.StakeDiffWindowSize
+// value; if the float precision is greater than zero. The precision is equal to
+// zero only when the block height value is divisible by the window size.
```

Comment:
It is good to have these kinds of comments on the implementation, but they should go inside the function.  The comment that begins with `// FunctionName` is special because it is used to describe the function's purpose and the meaning of the parameters to potential consumers of the function.  When reading https://godoc.org/github.com/decred/dcrdata/api/types, you need to think if the comment on the function is helpful in the context of API documentation.